pkg/controllers/inflightchecks/nodeshape.go

```diff
@@ -37,46 +36,27 @@ func NewNodeShape(provider cloudprovider.CloudProvider) Check {
 	}
 }
 
-func (n *NodeShape) Check(ctx context.Context, node *v1.Node, provisioner *v1alpha5.Provisioner, pdbs *deprovisioning.PDBLimits) ([]Issue, error) {
-	// ignore nodes that are deleting
-	if !node.DeletionTimestamp.IsZero() {
+func (n *NodeShape) Check(ctx context.Context, node *v1.Node, machine *v1alpha5.Machine, provisioner *v1alpha5.Provisioner, pdbs *deprovisioning.PDBLimits) ([]Issue, error) {
+	// ignore machines that are deleting
+	if !machine.DeletionTimestamp.IsZero() {
 		return nil, nil
 	}
-	// and nodes that haven't initialized yet
-	if node.Labels[v1alpha5.LabelNodeInitialized] != "true" {
+	// and machines that haven't initialized yet
+	if machine.StatusConditions().GetCondition(v1alpha5.MachineInitialized).IsTrue() {
 		return nil, nil
 	}
-
-	instanceTypes, err := n.provider.GetInstanceTypes(ctx, provisioner)
-	if err != nil {
-		return nil, err
-	}
-
-	instanceType, ok := lo.Find(instanceTypes, func(it *cloudprovider.InstanceType) bool { return it.Name == node.Labels[v1.LabelInstanceTypeStable] })
```

Comment:
No, it's because the `status.allocatable` represents the expected capacity of the Machine, we are no longer having to got to the cloudprovider to get the expected capacity